apps/api/src/app/analytics/analytics.module.ts

```diff
@@ -1,12 +1,12 @@
 import { Module } from '@nestjs/common';
-import { AnalyticsService } from '@novu/application-generic';
 import { HttpModule } from '@nestjs/axios';
 import { AnalyticsController } from './analytics.controller';
 import { HubspotIdentifyFormUsecase } from './usecases/hubspot-identify-form/hubspot-identify-form.usecase';
+import { SharedModule } from '../shared/shared.module';
 
 @Module({
-  imports: [HttpModule],
+  imports: [SharedModule, HttpModule],
   controllers: [AnalyticsController],
-  providers: [AnalyticsService, HubspotIdentifyFormUsecase],
+  providers: [HubspotIdentifyFormUsecase],
```

Comment:
```suggestion
  providers: [analyticsService, HubspotIdentifyFormUsecase],
```

Reply:
feels like you were missing the process.env.SEGMENT_TOKEN injection? was there something else?

Reply:
It's already imported under the SharedModule, so I think I would love to have a single instance of it

Reply:
It's confusing, but that's how all of our service are exported from app generic

![CleanShot 2024-10-13 at 17 41 30@2x](https://github.com/user-attachments/assets/bea166fb-d05e-4fb6-b473-9ccf74de748f)

Reply:
got it, i was not sure if it would create 2 instances.